lib/srv/alpnproxy/kube.go

```diff
@@ -121,6 +124,9 @@ func (m *KubeMiddleware) CheckAndSetDefaults() error {
 	if m.logger == nil {
 		m.logger = logrus.WithField(teleport.ComponentKey, "local_proxy_kube")
 	}
+	if m.context == nil {
+		m.context = context.Background()
+	}
```

Comment:
Can we always require the context?
It seems to be important for a clean shutdown, and using `context.Background()` is not very useful for that.

Reply:
Done https://github.com/gravitational/teleport/pull/43374/commits/9241628afbf2cb661b48a5bd8a7e1221fc447fca